packages/contentful-app-extensions/membership-reference/src/locations/Field.tsx

```diff
@@ -20,41 +20,72 @@ import {
 import { FieldExtensionSDK, Entry } from '@contentful/app-sdk';
 import { useSDK, useAutoResizer } from '@contentful/react-apps-toolkit';
 
+const capitalizeFirstLetter = (str: string) =>
+  str.slice(0, 1).toUpperCase() + str.slice(1);
+
 type CustomEntryCardProps = CustomEntityCardProps & { entity: Entry };
 type CardProps = {
   actions: EntryCardProps['actions'];
   onClick: CustomEntityCardProps['onEdit'];
   status: EntityStatus;
 };
-type TeamCardProps = {
+type MembershipCardProps = {
+  entityName: string;
+  showUserEmail: boolean;
   id: string;
-  role: string;
+  role?: string;
+  workstreamRole?: string;
   inactiveSince?: string;
 } & CardProps;
 
-const TeamCard = ({
+const MembershipCard = ({
+  entityName,
+  showUserEmail,
   id,
   actions,
   inactiveSince,
   role,
+  workstreamRole,
   onClick,
   status,
-}: TeamCardProps) => {
-  const { data: team } = useEntity<Entry>('Entry', id);
-  if (!team) {
+}: MembershipCardProps) => {
+  const { data } = useEntity<Entry>('Entry', id);
+  if (!data) {
     return <EntryCard isLoading />;
   }
+
+  const getEntityDisplayName = () => {
+    switch (entityName) {
+      case 'user':
+        return `${data.fields.firstName?.['en-US']} ${data.fields.lastName?.['en-US']}`;
+      case 'team':
+        return data.fields.displayName?.['en-US'];
+      case 'contributingCohort':
+        return data.fields.name?.['en-US'];
+      default:
+        throw new Error('Unknown entity');
+    }
+  };
+
   return (
     <EntryCard
-      contentType="Team"
+      contentType={capitalizeFirstLetter(entityName)}
       actions={actions}
       onClick={onClick}
       status={status}
     >
-      <Heading marginBottom="none">
-        {team.fields.displayName?.['en-US']}
-      </Heading>
-      <Paragraph marginBottom="none">{role}</Paragraph>
+      <Heading marginBottom="none">{getEntityDisplayName()}</Heading>
+      {entityName === 'user' &&
+        showUserEmail &&
+        data.fields.email?.['en-US'] && (
+          <Paragraph marginBottom="none">
+            {data.fields.email?.['en-US']}
+          </Paragraph>
+        )}
```

Comment:
@dianaguimaraes the issue was that I was trying to get the email from the membership entry before not the user entry 😛 thanks for the review 😄 

I've added this boolean parameter `showUserEmail` with default value `false` so we only need to add it to the fields where we want to show the email.
<img width="1440" alt="Screenshot 2023-07-26 at 13 56 39" src="https://github.com/yldio/asap-hub/assets/16595804/5e49d591-22f7-4c51-879b-6d428165b4f3">

So I added this param to the migrations `project` and `working group` below 

```
  const projects = migration.editContentType('projects');
  projects.changeFieldControl('members', 'app', 'Yp64pYYDuRNHdvAAAJPYa', {
    entityName: 'user',
    showUserEmail: true,     <------------
    showLinkEntityAction: false,
    showCreateEntityAction: true,
  });
};
```

```
  const workingGroups = migration.editContentType('workingGroups');
  workingGroups.changeFieldControl('members', 'app', 'Yp64pYYDuRNHdvAAAJPYa', {
    entityName: 'user',
    showUserEmail: true,     <------------
    showLinkEntityAction: false,
    showCreateEntityAction: true,
  });
```